scalableminds/webknossos
# Ensure unique volume layer names when changing names in frontend

This PR makes it impossible for the user to set layer names of volume annotation layers that already exist. The reason is to prevent duplicate names. The prevention is done when changing a name of a layer and when adding a new volume annotation layer. This pr does not change the names of existing volume layers but it marks them red if there are duplicates and asks the user to rename the layer with duplicated names.

### URL of deployed dev instance (used for testing):
- https://uniqueannotationlayernames.webknossos.xyz

### Steps to test:
I suggest testing this locally as the first step required is not (easily) possible on a dev branch.
- Create a volume annotation with multiple layers. Give the layers the same name. Add another layer that includes a `/` in its name.
- Now checkout the new branch, there layer names in the layer settings should now be red and a tooltip should say that these layers should be renamed to avoid duplicated names. The layer with the `/` should say that this character is not allowed and it should also be marked red.
- Try renaming the layers. This should work. Trying to enter a name that already exists should not be possible. Instead, an error toast should pop up. Entering a name that contains a `/` should also result in an error toast.
- Rename the layer so no names are duplicated and no name contains a `/`. This should then work.
- Try adding a new annotation layer. The suggested name should not match any of the existing names. Entering an existing name or one containing a `/` and then trying to confirm should result in an error toast.

### Issues:
- fixes [#6251](https://github.com/scalableminds/webknossos/issues/6251)

------
(Please delete unneeded items, merge only when none are left open)
